src/twisted/web/_newclient.py

```diff
@@ -813,7 +813,14 @@ def writeTo(self, transport):
             else:
                 return self._writeToContentLength(transport)
         else:
-            self._writeHeaders(transport, None)
+            # If the method semantics anticipate a body, include a
```

Comment:
> I guess what's true now is that sometimes a Content-Length headers is written by something other tyhan _writeToContentLength ... maybe that's what you mean. And, yea, I can see how that might confuse someone at some point.

Yep, that's it!  There are two things that bother me:

1. `_newclient` passes part of the HTTP wire protocol to private methods (`TEorCL`).  I can see why it does this, and the protocol format hasn't changed in 20 years, and it has nothing to do with this PR, so it's a lesser issue :)
2. `Content-Length` can be written outside of the method that says `_writeToContentLength`.  It's true that `_writeToContentLength` assumes a lot of things - most importantly that there's a body producer - so I thought maybe it could be given a name that reflected its actual behavior.

These are nits, though.